src/hotspot/share/oops/method.hpp

```diff
@@ -98,7 +98,22 @@ class Method : public Metadata {
   JFR_ONLY(DEFINE_TRACE_FLAG;)
 
 #ifndef PRODUCT
-  int               _compiled_invocation_count;  // Number of nmethod invocations so far (for perf. debugging)
+  union {
+    int64_t _compiled_invocation_count64;
+#if defined(VM_LITTLE_ENDIAN)
+    struct {
+      int     _compiled_invocation_count;  // Number of nmethod invocations so far (for perf. debugging)
+                                           // Must preserve this as int. Is used outside the jdk by SA.
```

Comment:
Why not update the SA code to access 64 bit?

Reply:
Two reasons: scope limitation and lack of expertise in SA code. Don't we need the 32bit decl anyway because it is contained in vmstructs.cpp?